### ISSUE TITLE
Memory read results in unexpected zero value.

### ISSUE THREAD

**Issue by nulldatamap** [kirkedal/compSys-e2017-sim#24]
The following example:
```asm
start:
  movq $0, %rax
  movq mat1(%rax), %rax
  movq $0, %rax
  movq mat2(%rax), %rax
  hlt

.pos 0x1000
mat1:
  .rand 1234 4

.pos 0x2000
mat2:
  .rand 1234 4
.pos 0x3000
out:
```
Results in `%rax` first loads to a non-zero value (as expected), but the second load results in a zero value.

`asm` output:
```asm
0x00000                               | start:
0x00000             30 00 00 00 00 00 | 		movq $0, %rax
0x00006             50 00 00 10 00 00 | 		movq mat1(%rax), %rax
0x0000C             30 00 00 00 00 00 | 		movq $0, %rax
0x00012             50 00 00 20 00 00 | 		movq mat2(%rax), %rax
0x00018                            00 | 		hlt
0x01000                               | mat1:
0x01000       0C BB 00 00 00 00 00 00 | 		.quad 47884
0x01008       0B E1 00 00 00 00 00 00 | 		.quad 57611
0x01010       01 92 00 00 00 00 00 00 | 		.quad 37377
0x01018       AD 4B 00 00 00 00 00 00 | 		.quad 19373
0x02000                               | mat2:
0x02000       0C BB 00 00 00 00 00 00 | 		.quad 47884
0x02008       0B E1 00 00 00 00 00 00 | 		.quad 57611
0x02010       01 92 00 00 00 00 00 00 | 		.quad 37377
0x02018       AD 4B 00 00 00 00 00 00 | 		.quad 19373
0x03000                               | out:
```

`sim -diff` output:
```asm
Registers:

  rax :                0
  rcx :                0
  rdx :                0
  rbx :                0
  rsi :                0
  rdi :                0
  rsp :                0
  rbp :                0
  r8  :                0
  r9  :                0
  r10 :                0
  r11 :                0
  r12 :                0
  r13 :                0
  r14 :                0
  r15 :                0

Instruction Pointer:

  rip :          0x00019

Changes to Memory:
```
I've inspected the execution using `-c`, and `%rax` gets directly set to zero on the memory load.

Assembled and simulated using 112da37f2ac .

**Comment by finnschiermer** [comment]
This is indeed very interesting. Thank you for reporting it. We'll get back.

**Comment by finnschiermer** [comment]
Should be fixed by #25
